cmd/kubeadm/app/util/runtime/runtime.go

```diff
@@ -128,16 +163,13 @@ func (runtime *CRIRuntime) RemoveContainers(containers []string) error {
 }
 
 // PullImage pulls the image
-func (runtime *CRIRuntime) PullImage(image string) error {
-	var err error
-	var out []byte
+func (runtime *CRIRuntime) PullImage(image string) (err error) {
 	for i := 0; i < constants.PullImageRetry; i++ {
-		out, err = runtime.crictl("pull", image).CombinedOutput()
-		if err == nil {
+		if _, err = runtime.impl.PullImage(context.Background(), runtime.imageService, &runtimeapi.ImageSpec{Image: image}, nil, nil); err == nil {
 			return nil
 		}
 	}
-	return errors.Wrapf(err, "output: %s, error", out)
+	return err
```

Comment:
previously this included any error that crictl may output on stderr/out, does it include which image failed exactly or do we need to:

```suggestion
	return errors.Wrapf(err, "failed to pull image %s", image)
```